frontend/src/components/account/ChangePassword/ChangePasswordForm.jsx

```diff
@@ -1,42 +1,143 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Account from '../Account';
 import styles from './ChangePasswordForm.module.scss';
 import Button from '../../ui-kit/components/Button';
 import { useNavigate } from 'react-router-dom';
+/* eslint-disable max-len */
+import { changePassword } from '../../../services/userData/changePassword.service';
+import ReactPasswordChecklist from 'react-password-checklist';
 
 const ChangePasswordForm = () => {
+  const [oldPassword, setOldPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmNewPassword, setConfirmNewPassword] = useState('');
+  const [oldPasswordError, setOldPasswordError] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+  const [showPasswordReq, setShowPasswordReq] = useState(true);
+  const [submitted, setSubmitted] = useState(false);
   const navigate = useNavigate();
 
   const NavigateAccount = () => {
     navigate('/account');
   };
 
+  const validatePassword = (value) => {
+    const passwordPattern = /^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[^\s]).{8,}$/;
+    return passwordPattern.test(value);
+  };
+
+  const handlePasswordChange = async (e) => {
+    e.preventDefault();
+    setSubmitted(true);
+
+    if (!oldPassword) {
+      setOldPasswordError('Введіть пароль');
+      return;
+    }
+    if (newPassword !== confirmNewPassword) {
+      setConfirmPasswordError('Паролі не співпадають');
+      return;
+    } else {
+      setConfirmPasswordError('');
+    }
+    if (validatePassword(newPassword)) {
+      const response = await changePassword({
+        email: localStorage.getItem('email'),
+        oldPassword,
+        newPassword,
+      });
+      if (response.success) {
+        navigate('/account');
+      } else if (response.error === 'Невірний пароль!') {
+        setOldPasswordError(response.error);
+      }
+    }
+  };
+
   return (
     <div>
       <Account title="Редагування особистої інформації">
         <div className={styles.formContainer}>
           <h2 className={styles.title}> Зміна паролю</h2>
-          <form action="">
+          <form onSubmit={handlePasswordChange}>
+            <div className={styles.passwordContainer}>
+              <label htmlFor="password">Введіть старий пароль</label>
+              <input
+                className={styles.dataInput}
+                id="oldPassword"
```

Comment:
Do you think there's a need to use controlled input here? As I understand it, you only need this info on submission. Maybe it's better to use ref here

Reply:
done